### DISCUSSION TITLE
How to fix hydration issue of Microfrontend(Nextjs) + SSR using dynamic import {ssr: true}

### DISCUSSION THREAD

**Discussion by gskaushal92** [vercel/next.js#51809] [Help]
### Summary

I have created remote had consumer app using nextjs. (We can have remote app with create-react-app as well)
I am using module federation for microfrontend.
I am using dynamic import to use remote app modules in consumer app with {ssr: true}.
I have deployed remote and consumer app separately.

Now whenever I update the html of remote module and deploy it. Then on opening the consumer application. I am getting following common error "hydration failed because the initial UI does not match" .

These are my remote and host app.
remote app : https://microfrontend-next-poc-app1.vercel.app/
host app: https://microfrontend-next-poc-host-app.vercel.app/

In host app , I am getting hydration issue in console. also If I check the view source. I will not get the updated html.

Is there a way to resolve this issue with {ssr: true}. So that my SEO should not impact.

### Additional information

```js
My Use case is that, I want all mfe can be independently deployed and there should not be the need to deploy the container app whenever any mfe is updated and deployed.
But changes should reflect automatically on run time without any hydration issue and also maintain the SEO compatibility.
```

### Example

https://microfrontend-next-poc-host-app.vercel.app/

**Comment by gskaushal92** [comment]
this issue got resolved now. It's just that container app was taking time to reflect the changes.

**Reply by Ganesh1991** [reply]
Did you use the NextJs App router while implementing the micro frontend?
